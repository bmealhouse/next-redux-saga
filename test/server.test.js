/** @jest-environment node */
import withRedux from 'next-redux-wrapper'
import withReduxSaga from '..'
import AsyncGetInitialProps from './components/async-get-initial-props'
import ClassComponent from './components/class-component'
import FunctionalComponent from './components/functional-component'
import SyncGetInitialProps from './components/sync-get-initial-props'
import configureStore from './store/configure-store'
import createSnapshot from './utils/create-snapshot'
import getInitialProps from './utils/get-initial-props'
import {
  ASYNC_REDUX_SAGA_PROP_TEXT,
  STATIC_PROP_TEXT,
  SYNC_REDUX_PROP_TEXT,
} from './constants'

test('Wrapped component passes along React props', () => {
  const WrappedComponent = withRedux(configureStore)(
    withReduxSaga(FunctionalComponent),
  )

  createSnapshot(WrappedComponent)
})

test('Wrapped component skips getInitialProps when it does not exist', async () => {
  const WrappedComponent = withRedux(configureStore)(
    withReduxSaga(ClassComponent),
  )

  const props = await getInitialProps(WrappedComponent)
  expect(props.isServer).toBeTruthy()

  createSnapshot(WrappedComponent, props)
})

test('Wrapped component awaits synchronous getInitialProps', async () => {
  const WrappedComponent = withRedux(configureStore)(
    withReduxSaga(SyncGetInitialProps),
  )

  const props = await getInitialProps(WrappedComponent)
  expect(props.isServer).toBeTruthy()
  expect(props.initialState).toEqual({syncReduxProp: SYNC_REDUX_PROP_TEXT})
  expect(props.initialProps).toEqual({staticProp: STATIC_PROP_TEXT})

  createSnapshot(WrappedComponent, props)
})

test('Wrapped component awaits asynchronous getInitialProps', async () => {
  const WrappedComponent = withRedux(configureStore)(
    withReduxSaga(AsyncGetInitialProps),
  )

  const props = await getInitialProps(WrappedComponent)
  expect(props.isServer).toBeTruthy()
  expect(props.initialState).toEqual({
    syncReduxProp: SYNC_REDUX_PROP_TEXT,
    asyncReduxSagaProp: ASYNC_REDUX_SAGA_PROP_TEXT,
  })
  expect(props.initialProps).toEqual({staticProp: STATIC_PROP_TEXT})

  createSnapshot(WrappedComponent, props)
})
