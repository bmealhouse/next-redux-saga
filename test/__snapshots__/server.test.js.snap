// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapped component awaits asynchronous getInitialProps 1`] = `
<Connect(AsyncGetInitialProps)
  isServer={true}
  staticProp="Static message from getInitialProps()"
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "runSagaTask": [Function],
      "sagaTask": Object {
        "@@redux-saga/TASK": true,
        "cancel": [Function],
        "cont": [Function],
        "context": Object {},
        "end": [Function],
        "error": [Function],
        "id": 19,
        "isAborted": [Function],
        "isCancelled": [Function],
        "isRoot": true,
        "isRunning": [Function],
        "joiners": null,
        "meta": Object {
          "location": undefined,
          "name": "rootSaga",
        },
        "queue": Object {
          "abort": [Function],
          "addTask": [Function],
          "cancelAll": [Function],
          "getTasks": [Function],
        },
        "result": [Function],
        "setContext": [Function],
        "toPromise": [Function],
      },
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
/>
`;

exports[`Wrapped component awaits synchronous getInitialProps 1`] = `
<Connect(SyncGetInitialProps)
  isServer={true}
  staticProp="Static message from getInitialProps()"
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "runSagaTask": [Function],
      "sagaTask": Object {
        "@@redux-saga/TASK": true,
        "cancel": [Function],
        "cont": [Function],
        "context": Object {},
        "end": [Function],
        "error": [Function],
        "id": 13,
        "isAborted": [Function],
        "isCancelled": [Function],
        "isRoot": true,
        "isRunning": [Function],
        "joiners": null,
        "meta": Object {
          "location": undefined,
          "name": "rootSaga",
        },
        "queue": Object {
          "abort": [Function],
          "addTask": [Function],
          "cancelAll": [Function],
          "getTasks": [Function],
        },
        "result": [Function],
        "setContext": [Function],
        "toPromise": [Function],
      },
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
/>
`;

exports[`Wrapped component passes along React props 1`] = `
<Connect(FunctionalComponent)
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "runSagaTask": [Function],
      "sagaTask": Object {
        "@@redux-saga/TASK": true,
        "cancel": [Function],
        "cont": [Function],
        "context": Object {},
        "end": [Function],
        "error": [Function],
        "id": 1,
        "isAborted": [Function],
        "isCancelled": [Function],
        "isRoot": true,
        "isRunning": [Function],
        "joiners": Array [],
        "meta": Object {
          "location": undefined,
          "name": "rootSaga",
        },
        "queue": Object {
          "abort": [Function],
          "addTask": [Function],
          "cancelAll": [Function],
          "getTasks": [Function],
        },
        "result": [Function],
        "setContext": [Function],
        "toPromise": [Function],
      },
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
/>
`;

exports[`Wrapped component skips getInitialProps when it does not exist 1`] = `
<Connect(ClassComponent)
  isServer={true}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "runSagaTask": [Function],
      "sagaTask": Object {
        "@@redux-saga/TASK": true,
        "cancel": [Function],
        "cont": [Function],
        "context": Object {},
        "end": [Function],
        "error": [Function],
        "id": 7,
        "isAborted": [Function],
        "isCancelled": [Function],
        "isRoot": true,
        "isRunning": [Function],
        "joiners": null,
        "meta": Object {
          "location": undefined,
          "name": "rootSaga",
        },
        "queue": Object {
          "abort": [Function],
          "addTask": [Function],
          "cancelAll": [Function],
          "getTasks": [Function],
        },
        "result": [Function],
        "setContext": [Function],
        "toPromise": [Function],
      },
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
/>
`;
